src/sdk/pynni/nni/compression/torch/pruning/structured_pruning.py

```diff
@@ -129,11 +281,46 @@ def get_mask(self, base_mask, weight, num_prune, wrapper, wrapper_idx):
 
         return {'weight_mask': mask_weight.detach(), 'bias_mask': mask_bias}
 
+
+class L1ConstrainedFilterPrunerMasker(ConstrainedStructuredWeightMasker):
+
+    def get_mask(self, wrapper, wrapper_idx, channel_mask):
+        weight = wrapper.module.weight.data
+        filters = weight.shape[0]
+        w_abs_structured = self._get_channel_sum(wrapper, wrapper_idx)
+        # set the sum of the already pruned channel to
+        # zero, so that these channel will be pruned.
+        w_abs_structured = w_abs_structured * channel_mask
+        sparsity = wrapper.config['sparsity']
+        num_prune = int(filters * sparsity)
+        if num_prune > 0:
+            threshold = torch.topk(
```

Comment:
better to merge constrained masker class and existing masker class to eliminate duplicated algo code.